api/src/main/java/org/xwiki/apps/googleapps/GoogleAppsManager.java

```diff
@@ -0,0 +1,242 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.apps.googleapps;
+
+import java.io.IOException;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+import org.xwiki.component.annotation.Role;
+import org.xwiki.stability.Unstable;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.services.drive.model.File;
+import com.google.api.services.drive.model.FileList;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+
+/**
+ * The specification of the methods that the manager of the GoogleApps application
+ * is doing. Methods of this interface are mostly called by the script-service (itself
+ * called by the views).
+ *
+ * @version $Id$
+ * @since 3.0
+ */
+@Role
+public interface GoogleAppsManager
+{
+
+
+    /**
+     * @return if the application is licensed and activated
+     * @throws XWikiException in case a context cannot be read from thread.
+     * @since 3.0
+     */
+    @Unstable
+    boolean isActive() throws XWikiException;
```

Comment:
Oh, that is a big suggestion!
I've been trying to surf around and indeed this is done filemanager, for example.
So the solution is to remove any dependency on XWikiException within the API but within the implementation's methods. Right?

Reply:
If you separate the API and the implementation in two different maven modules then the implementation can depend on oldcore. Of course, the API can depend on oldcore too if it's really needed, but most of the time it can be avoided. So in your case, the cleanest is to create a ``GoogleAppsException`` and use it in the API. The implementation should normally catch **checked** exceptions, wrap them in ``GoogleAppsException`` and then re-throw.

Reply:
Implemented in a5a69382ef5d8946f3b83605b3fd82bd967fb5ec
paul